compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -0,0 +1,244 @@
+                                                   indices[indices.size() - 2],
+                                                   indices[indices.size() - 1]);
+            Value diag = b.create<arith::SelectOp>(isDiag, one, zero);
+            Value sub = b.create<arith::SubFOp>(diag, h);
+
+            b.create<linalg::YieldOp>(sub);
+          })
+      .getResult(0);
+}
+
+// Compute the householder matrice with the k-th column of matrix and k-th tau
+// value.
+static Value computeHouseholderSlice(Value matrix, Value tau, Value k,
+                                     ImplicitLocOpBuilder b) {
+  auto matrixTy = cast<ShapedType>(matrix.getType());
+  int rank = matrixTy.getRank();
+
+  llvm::SmallVector<OpFoldResult> vStrides(rank, b.getIndexAttr(1));
+  llvm::SmallVector<int64_t> vShape(matrixTy.getShape());
+  vShape[vShape.size() - 1] = 1;
+
+  llvm::SmallVector<OpFoldResult> vOffsets(rank, b.getIndexAttr(0));
+  vOffsets[vOffsets.size() - 1] = k;
+
+  llvm::SmallVector<OpFoldResult> vSizes;
+  for (auto v : vShape) {
+    vSizes.push_back(b.getIndexAttr(v));
+  }
+
+  auto sliceTy = RankedTensorType::get(vShape, matrixTy.getElementType());
+  Value v = b.create<tensor::ExtractSliceOp>(sliceTy, matrix, vOffsets, vSizes,
+                                             vStrides);
+
+  SmallVector<ReassociationIndices> reass;
+  for (int i = 0; i < rank - 2; ++i) {
+    reass.push_back({i});
+  }
+  reass.push_back({rank - 2, rank - 1});
+
+  ArrayRef<int64_t> collapseVShape(vShape.begin(), vShape.end() - 1);
+  auto collapseVTy =
+      RankedTensorType::get(collapseVShape, matrixTy.getElementType());
+  Value collapseV = b.create<tensor::CollapseShapeOp>(collapseVTy, v, reass);
+
+  Value householder = computeHouseholder(collapseV, tau, k, b);
+  return householder;
+}
+
+struct HouseholderReflectorRewriter final
+    : OpRewritePattern<mlir::stablehlo::CustomCallOp> {
+  using OpRewritePattern<mlir::stablehlo::CustomCallOp>::OpRewritePattern;
+  using OpAdaptor = mlir::stablehlo::CustomCallOp::Adaptor;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::CustomCallOp op,
+                                PatternRewriter &rewriter) const final {
+    if (op.getCallTargetName() != "ProductOfElementaryHouseholderReflectors") {
+      return rewriter.notifyMatchFailure(
+          op, "not ProductOfElementaryHouseholderReflectors");
+    }
+
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    auto matrix = op.getOperand(0);
+    auto tau = op.getOperand(1);
+    auto matrixTy = cast<ShapedType>(matrix.getType());
+    auto tauTy = cast<ShapedType>(tau.getType());
+    auto rank = matrixTy.getRank();
+
+    if (isa<ComplexType>(matrixTy.getElementType())) {
+      return rewriter.notifyMatchFailure(op, "complex types not supported");
+    }
+
+    if (rank < 2) {
+      return rewriter.notifyMatchFailure(op, "requires minimum rank 2 matrix");
+    }
+
+    if (!matrixTy.hasStaticShape() || !tauTy.hasStaticShape()) {
```

Comment:
What complicates this part? E.g., could we document missing here so that we don't forget to enable.

Reply:
Added a comment. It would make sense to have someone test / implement but for a first pass its easier to ignore. We can roll back once we start encountering more dynamic shape cases.